sentry/src/main/java/io/sentry/cache/EnvelopeCache.java

```diff
@@ -429,7 +431,9 @@ public void discard(final @NotNull SentryEnvelope envelope) {
   /** Awaits until the previous session (if any) is flushed to its own file. */
   public boolean waitPreviousSessionFlush() {
     try {
-      return previousSessionLatch.await(options.getFlushTimeoutMillis(), TimeUnit.MILLISECONDS);
+      // use fixed timeout instead of configurable options.getFlushTimeoutMillis() to ensure there's
+      // enough time to flush the session to disk
```

Comment:
@romtsn if you've better reasoning here, please adapt the comment 😊

Reply:
yeah, it's good!